version: '2.1'

services:
  db:
    image: postgres
    restart: always
    environment:
      - POSTGRES_DB=gitlab_bot
      - POSTGRES_USER=gitlab_bot
      - POSTGRES_PASSWORD=9_9
    volumes:
      - ./pgdata:/var/lib/postgresql/data
    ports:
      - "1001:5432"
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U gitlab_bot" ]
      interval: 1s
      timeout: 5s
      retries: 5
    networks:
      tgbot:
        aliases:
          - postgres

  migration:
    build:
      context: .
    restart: on-failure
    depends_on:
      - db
    command: "./migration"
    networks:
      - tgbot

  pgadmin:
    image: fenglc/pgadmin4:2.1-python2.7-alpine
    restart: always
    depends_on:
      - db
    ports:
      - "9999:5050"
    environment:
      DEFAULT_PASSWORD: gitlab_bot_pass
      DEFAULT_USER: gitlab_bot
    networks:
      - tgbot

  app:
    build:
      context: .
    restart: always
    ports:
      - "10010:10010"
    depends_on:
      - migration
    command: "./main"
    networks:
      tgbot:
        aliases:
          - backend
networks:
  tgbot:
    driver: bridge
    driver_opts:
      com.docker.network.enable_ipv6: "false"
